internal/provider/org/group_resource.go

```diff
@@ -0,0 +1,314 @@
+package org
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	govcdtypes "github.com/vmware/go-vcloud-director/v2/types/v56"
+
+	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/client"
+	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/helpers/stringpm"
+)
+
+var (
+	_ resource.Resource                = &orgGroupResource{}
+	_ resource.ResourceWithConfigure   = &orgGroupResource{}
+	_ resource.ResourceWithImportState = &orgGroupResource{}
+)
+
+// NewOrgGroupResource is a helper function to simplify the provider implementation.
+func NewOrgGroupResource() resource.Resource {
+	return &orgGroupResource{}
+}
+
+// orgGroupResource is the resource implementation.
+type orgGroupResource struct {
+	client *client.CloudAvenue
+}
+
+type orgResourceModel struct {
```

Comment:
Please change the name of the struct 

```suggestion
type orgGroupResourceModel struct {
```